postgres/datadog_checks/postgres/postgres.py

```diff
@@ -514,7 +514,10 @@ def _connect(self, host, port, user, password, dbname, ssl):
         else:
             if host == 'localhost' and password == '':
                 # Use ident method
-                self.db = psycopg2.connect("user=%s dbname=%s, application_name=%s" % (user, dbname, "datadog-agent"))
+                connection_string = "user=%s dbname=%s, application_name=%s" % (user, dbname, "datadog-agent")
+                if query_timeout:
+                    connection_string += "options='-c statement_timeout=%s'" % query_timeout
```

Comment:
I think you're missing a space before options.

Considering this is untested, pending a refactor, and possibly broken here, can we move that a separate PR? Thanks.

Reply:
I'd need at least the feature in dark mode for the tests to pass

Reply:
Fixed and added unit test